src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/commands.py

```diff
@@ -88,7 +89,8 @@ def transform_deployments_list(result):
 
 cli_command(__name__, 'group deployment create', 'azure.cli.command_modules.resource.custom#deploy_arm_template', no_wait_param='no_wait', exception_handler=handle_long_running_operation_exception)
 cli_generic_wait_command(__name__, 'group deployment wait', 'azure.mgmt.resource.resources.operations.deployments_operations#DeploymentsOperations.get', cf_deployments)
-cli_command(__name__, 'group deployment list', 'azure.mgmt.resource.resources.operations.deployments_operations#DeploymentsOperations.list_by_resource_group', cf_deployments, table_transformer=transform_deployments_list)
+cli_command(__name__, 'group deployment list', 'azure.mgmt.resource.resources.operations.deployments_operations#DeploymentsOperations.list_by_resource_group', cf_deployments, table_transformer=transform_deployments_list, resource_type=ResourceType.MGMT_RESOURCE_RESOURCES, min_api='2017-05-10')
+cli_command(__name__, 'group deployment list', 'azure.mgmt.resource.resources.operations.deployments_operations#DeploymentsOperations.list', cf_deployments, table_transformer=transform_deployments_list, resource_type=ResourceType.MGMT_RESOURCE_RESOURCES, max_api='2016-09-01')
```

Comment:
This simply looks like `group deployment list` was registered twice and makes it likely that this "duplicate entry" could get deleted in the future by mistake. Please use one of the already established mechanisms (VersionConstraint or supported_api_version) that more explicitly calls attention to the API version.  For this instance, I would recommend `supported_api_version`.
```Python
if supported_api_version(ResourceType.MGMT_RESOURCE_RESOURCE, min_api='...'):
   cli_command(...)
else:
   cli_command(...)
```